File: software/examples/rust/example_hello_world.rs
use std::{error::Error, io};

use tinkerforge::{ipconnection::IpConnection, oled_64x48_bricklet::*};

const HOST: &str = "127.0.0.1";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your OLED 64x48 Bricklet

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection
    let oled_64x48_bricklet = OLED64x48Bricklet::new(UID, &ipcon); // Create device object

    ipcon.connect(HOST, PORT).recv()??; // Connect to brickd
                                        // Don't use device before ipcon is connected

    // Clear display
    oled_64x48_bricklet.clear_display();

    // Write "Hello World" starting from upper left corner of the screen
    oled_64x48_bricklet.write_line(0, 0, "Hello World".to_string());

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_hello_world.go
package main

import (
	"fmt"
	"tinkerforge/ipconnection"
	"tinkerforge/oled_64x48_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your OLED 64x48 Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	oled, _ := oled_64x48_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Clear display
	oled.ClearDisplay()

	// Write "Hello World" starting from upper left corner of the screen
	oled.WriteLine(0, 0, "Hello World")

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
